test/parse_test.rb

```diff
@@ -1740,6 +1740,47 @@ class ParseTest < Test::Unit::TestCase
     assert_parses expected, "begin a; ensure b; end"
   end
 
+  test "parses empty hash" do
+    assert_parses HashNode(BRACE_LEFT("{"), [], BRACE_RIGHT("}")), "{}"
+  end
+
+  test "parses hash with hashrocket keys" do
+    expected = HashNode(
+      BRACE_LEFT("{"),
+      [AssocNode(
+        CallNode(nil, nil, IDENTIFIER("a"), nil, nil, nil, "a"),
+        EQUAL_GREATER("=>"),
```

Comment:
Do you think we should track this here?

I realized Ruby maybe doesn't keep track of the type of hash syntax you used in the parse tree,

```
ruby --dump=parsetree -e "{ a: b }"                                               11:28:48
###########################################################
## Do NOT use this node dump for any purpose other than  ##
## debug and research.  Compatibility is not guaranteed. ##
###########################################################

# @ NODE_SCOPE (id: 5, line: 1, location: (1,0)-(1,8))
# +- nd_tbl: (empty)
# +- nd_args:
# |   (null node)
# +- nd_body:
#     @ NODE_HASH (id: 4, line: 1, location: (1,0)-(1,8))*
#     +- nd_brace: 1 (hash literal)
#     +- nd_head:
#         @ NODE_LIST (id: 2, line: 1, location: (1,2)-(1,6))
#         +- nd_alen: 2
#         +- nd_head:
#         |   @ NODE_LIT (id: 1, line: 1, location: (1,2)-(1,4))
#         |   +- nd_lit: :a
#         +- nd_head:
#         |   @ NODE_VCALL (id: 0, line: 1, location: (1,5)-(1,6))
#         |   +- nd_mid: :b
#         +- nd_next:
#             (null node)
```

I am starting to think about how we would do `{ a: b }` and maybe it would be difficult to parse the `COLON` as its own token instead of being part of the `LABEL`/ key.

Maybe we can skip this and just track the key and value?